x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/servicenow_connectors.tsx

```diff
@@ -5,157 +5,176 @@
 
   useEffect(
-    () => setCallbacks({ beforeActionConnectorSave }),
-    [beforeActionConnectorSave, setCallbacks]
+    () => registerPreSubmitValidator(preSubmitValidator),
+    [preSubmitValidator, registerPreSubmitValidator]
   );
 
   const onMigrateClick = useCallback(() => setShowUpdateConnector(true), []);
   const onModalCancel = useCallback(() => setShowUpdateConnector(false), []);
 
-  const onUpdateConnectorConfirm = useCallback(async () => {
-    try {
-      await getApplicationInfo();
-
-      await updateActionConnector({
-        http,
-        connector: {
-          name: action.name,
-          config: { ...config, usesTableApi: false },
-          secrets: { ...secrets },
-        },
-        id: action.id,
-      });
-
-      editActionConfig('usesTableApi', false);
-      setShowUpdateConnector(false);
-
-      toasts.addSuccess({
-        title: i18n.UPDATE_SUCCESS_TOAST_TITLE(action.name),
-        text: i18n.UPDATE_SUCCESS_TOAST_TEXT,
-      });
-    } catch (err) {
-      /**
-       * getApplicationInfo may throw an error if the request
-       * fails or if there is a REST api error.
-       *
-       * We silent the errors as a callout will show and inform the user
-       */
-    }
-  }, [getApplicationInfo, http, action.name, action.id, secrets, config, editActionConfig, toasts]);
-
-  /**
-   * Defaults the usesTableApi attribute to false
-   * if it is not defined. The usesTableApi attribute
-   * will be undefined only at the creation of
-   * the connector.
-   */
-  useEffect(() => {
-    if (config.usesTableApi == null) {
-      editActionConfig('usesTableApi', false);
-    }
-  });
+  const onUpdateConnectorConfirm = useCallback(
+    async (updatedConnector: UpdateConnectorFormSchema['updatedConnector']) => {
+      const connectorToUpdate = {
+        name: name ?? '',
+        config: { ...updatedConnector.config, usesTableApi: false },
+        secrets: { ...updatedConnector.secrets },
+        id: id ?? '',
+      };
+
+      try {
+        await getApplicationInfo({
+          ...connectorToUpdate,
+          isDeprecated,
+          isPreconfigured: false,
+          actionTypeId,
+        });
+
+        const res = await updateActionConnector({
+          http,
+          connector: connectorToUpdate,
+          id: id ?? '',
+        });
+
+        toasts.addSuccess({
+          title: i18n.UPDATE_SUCCESS_TOAST_TITLE(name ?? ''),
+          text: i18n.UPDATE_SUCCESS_TOAST_TEXT,
+        });
+
+        setShowUpdateConnector(false);
+
+        updateFieldValues({
+          isDeprecated: res.isDeprecated,
+          config: updatedConnector.config,
+        });
+      } catch (err) {
+        setUpdateErrorMessage(err.message);
+      }
+    },
+    [name, id, getApplicationInfo, isDeprecated, actionTypeId, http, updateFieldValues, toasts]
+  );
 
   return (
     <>
-      {showUpdateConnector && (
+      {actionTypeId && showUpdateConnector && (
```

Comment:
There's no chance that `action.actionTypdId` would be null or undefined separately from `actionTypeId` right? Otherwise we'd need to check both.